nncf/extensions/src/quantization/cuda/functions_cuda_impl.cu

```diff
@@ -39,6 +39,7 @@ ScaleType get_scale_type(const at::Tensor& input, const at::Tensor& input_low, c
             TORCH_CHECK(input_range.size(1) == scale_count, "Scale shape is not flat");
             return  ScaleType::PER_ACTIVATION_CHANNEL;
         }
+        return ScaleType::PER_WEIGHT_CHANNEL;
```

Comment:
Shouldn't skip the TORCH_CHECKS like that. Please add ` || scale_count == 1` in the condition on line 30 instead.

Reply:
Update to a discussed version